extensions/schema-compare/src/schemaCompareMainWindow.ts

```diff
@@ -285,14 +286,21 @@ export class SchemaCompareMainWindow {
 	}
 
 	// only for test
-	public getDeploymentOptions(): mssql.DeploymentOptions {
-		return this.deploymentOptions;
-	}
-
-	// only for test
-	public verifyButtonsState(generateScriptButtonState: boolean, applyButtonState: boolean): boolean {
+	public verifyButtonsState(compareButtonState: boolean, optionsButtonState: boolean, switchButtonState: boolean,
```

Comment:
Rather than verify I would suggest having a get button state and return a button state object and then verfy that in test code?  This seems like test in product code

Reply:
Or create a Test version of the class that extends this class and adds this additional test functionality.

Reply:
Updated to have a test version of the class and added test specific functionality in it.